### PR TITLE
5395 nan appears in datagrid fields for undefined selections

### PR DESCRIPTION
**Pull request by Jah25**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Updated the Decimal formatter to prevent NaN from being displayed when the user doesn't enter a value and updated the placeholder example page dropdown so that the blank value was not a number which caused other options to display NaN when selected. Additionally added more type safety to coercion to prevent edge cases like the drop down from causing future issues

**Related github/jira issue (required)**:

Closes ([#5395](https://github.com/infor-design/enterprise/issues/5395))

**Steps necessary to review your pull request (required)**:

-Build and Run the application
- navigate to the datagrid placeholder example http://localhost:4000/components/datagrid/example-placeholder.html

To test the price issue
- click into the price column of row one
- click outside the cell
- no NaN should appear in the field

To test the date issue
-select the Action dropdown in row one
-select the blank option
-select the Action dropdown again
-select On Hold
-the field should display On Hold instead of NaN

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Comment by Jah25** [comment]
@tmcconechy I'm a little confused by the linting error here, I'm not seeing it locally and this doesn't seem like unnecessary escaping since it's a regex symbol escape.

**Comment by tmcconechy** [comment]
@Jah25 i'm seeing it when i run `npm run lint:ci` also i see it in visual code. 

```
tmcconechy@usmvtmcconechy enterprise2 % npm run lint:ci
....

/Users/tmcconechy/Dev/enterprise2/src/components/datagrid/datagrid.js
  10355:56  error  Unnecessary escape character: \[  no-useless-escape
  10355:71  error  Unnecessary escape character: \/  no-useless-escape
 ```
 
Would suggest installing the plugins as you can see it while you edit. 
 Screen Shot 2021-07-19 at 12 42 29 PM

These are what i have enabled: (I think its all the "lint" ones and "eslint")

Screen Shot 2021-07-19 at 12 43 20 PM
Screen Shot 2021-07-19 at 12 43 28 PM


But ultimately for this one i think you can ignore the rule as its valid in this case.
```js
    // eslint-disable-next-line no-useless-escape
    const nonNumberCharacters = /[A-Za-z!@#$%^&*()_+\-=\[\]{};':"\\|<>\/?]/;
```

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
 - `[Multiselect]` Fixed a regression bug where clear icon were not properly aligned on compact mode. ([#5396](https://github.com/infor-design/enterprise/issues/5396))
 - `[Popdown]` Remove deprecation console warning. We still consider this component deprecated but will not remove until 5.0 version. The warning was only removed for now. ([#1070](https://github.com/infor-design/enterprise-ng/issues/1070))
 -`[Datagrid]` Fixed a bug where the font color on tags was black when a row was hovered over in dark mode. Font color now white. ([#5289](https://github.com/infor-design/enterprise/issues/5289))
+-`[Datagrid]` Fixed issues with NaN displaying on Decimal and Dropdown inputs when blank options are selected. ([#5395] (https://github.com/infor-design/enterprise/issues/5395))
```
Comment (tmcconechy) [inline docs/CHANGELOG.md:25]:
```suggestion
-`[Datagrid]` Fixed issues with NaN displaying on Decimal and Dropdown inputs when blank options are selected. ([#5395](https://github.com/infor-design/enterprise/issues/5395))
```

Will fix the linter (link is malformed)

**Comment by Jah25** [comment]
@tmcconechy I had a question about the test this is failing on, I have a fix for it that replaces the NaN check with a regex but I was wondering what exactly this test is for since $#,##0.00 seems to be a desired outcome for the Decimal filter.

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,7 @@
 - `[Multiselect]` Fixed a regression bug where clear icon were not properly aligned on compact mode. ([#5396](https://github.com/infor-design/enterprise/issues/5396))
 - `[Popdown]` Remove deprecation console warning. We still consider this component deprecated but will not remove until 5.0 version. The warning was only removed for now. ([#1070](https://github.com/infor-design/enterprise-ng/issues/1070))
 -`[Datagrid]` Fixed a bug where the font color on tags was black when a row was hovered over in dark mode. Font color now white. ([#5289](https://github.com/infor-design/enterprise/issues/5289))
+-`[Datagrid]` Fixed issues with NaN displaying on Decimal and Dropdown inputs when blank options are selected. ([#5395](https://github.com/infor-design/enterprise/issues/5395))
```
Comment (tmcconechy) [inline docs/CHANGELOG.md:25]: FYI we usually keep this in alphabetical order (i see some werent yours). We can fix that later tho so just a mention.
Or go ahead and adjust them all.

**Review by tmcconechy — APPROVED** [review]
Nice job on this one @Jah25 this looks like a great fix 👍🏻

File: docs/CHANGELOG.md (hunk 2 above)
Comment (Jah25) [reply docs/CHANGELOG.md:25]: Oh sorry about that! I thought they were in order of fix, I'll make sure to fix it on the next PR.

Comment (tmcconechy) [reply docs/CHANGELOG.md:25]: No problem just letting you know. But somehow they got out of order. Not your fault

**Review by ericangeles — APPROVED** [review]

**Pull request merged**